### PR TITLE
Add new option - minimize to support outputting minimized css file

### PR DESCRIPTION
**Pull request by tom76kimo**
@src-code @roderickhsiao 

- Add support of outputting minimized css file
- Added support of processing output css file with `postcss`
- Change to async loader
- Upgrade `react` version

### Usage
add `minimize: true` to loader query
add `postcssPlugins` as an array with needed plugins
```js
var autoprefixer = require('autoprefixer');
.
.
.

{
    test: /\.jsx?$/,
    exclude: /(node_modules)/,
    loader: '../dist/atomicLoader',
    query: {
        postcssPlugins: [autoprefixer]
        minimize: true,
        configPath: [
            path.resolve('./atomCssConfig.js'),
            path.resolve('./atomCssConfig2.js')
        ]
    }
}
```

### CODE REVIEW

File: lib/atomicLoader.js [hunk 1]
```diff
@@ -48,51 +52,75 @@ let configObject = {
...
+            }
+        }
 
-    return;
-}
+        const finalConfig = atomizer.getConfig(foundClasses, pathConfig.configs || {});
+        const cssString = atomizer.getCss(finalConfig, pathConfig.options || {});
+
+        if (minimize) {
+            const pipeline = postcss(cssnano())
```
Comment (roderickhsiao) [inline lib/atomicLoader.js]: should we allow user to pass minified options to `cssnano`?

Comment (roderickhsiao) [reply lib/atomicLoader.js]: another thought will be allowing user to pass in different `postcss` plugin themself (either a whitelisted plugin or just for all plugin) for easy integration. how do you think?

File: package.json [hunk 2]
```diff
@@ -1,34 +1,36 @@
...
+        "postcss": "^6.0.12"
+    },
+    "devDependencies": {
+        "babel-cli": "^6.6.5",
+        "babel-core": "^6.7.2",
+        "babel-loader": "^6.2.4",
+        "babel-preset-es2015": "^6.6.0",
+        "babel-preset-react": "^6.5.0",
+        "chai": "^3.2.0",
+        "react": "^0.14.7",
```
Comment (roderickhsiao) [inline package.json]: so old the react and webpack version btw :)

File: lib/atomicLoader.js (hunk 1 above)
Comment (tom76kimo) [reply lib/atomicLoader.js]: I was thinking of making it as easy to use as possible. But, well, maybe user needs more flexible capabilities. I think I can make changes to support plugins.

File: package.json (hunk 2 above)
Comment (tom76kimo) [reply package.json]: Great catch 👍

File: lib/atomicLoader.js (hunk 1 above)
Comment (tom76kimo) [reply lib/atomicLoader.js]: However, after reviewing [optimisation options](http://cssnano.co/guides/optimisations/) and consider the css file which generate by `atomizer`, I think the default supports of `cssnano` is enough for the purpose we want to achieve.

**Comment by tom76kimo** [comment]
Added support of processing output css file with postcss

File: lib/atomicLoader.js (hunk 1 above)
Comment (tom76kimo) [reply lib/atomicLoader.js]: Updated!

**Comment by roderickhsiao** [comment]
👍 thanks

**Comment by roderickhsiao** [comment]
probably add readme will be good

**Comment by tom76kimo** [comment]
Updated!

**Pull request merged**